src/oauthHandlers/tokenHandlerStrategyClasses/tokenHandlerClient.js

```
@@ -159,7 +159,7 @@ class TokenHandlerClient {
           );
           for (let key in decodedLaunch) {
             if (!responseBody[key]) {
-              responseBody[key] = decodedLaunch[key];
+              responseBody[key] = String(decodedLaunch[key]);
```

Modifying this section (and actually line 154 above which also would require similar casting) only resolves the /token response payload, however subsequent fetches during token validation would still result in possible integer launch context.

Instead, we want to shift this validation left prior to the saving of the launch context in the first place.

https://github.com/department-of-veterans-affairs/lighthouse-oauth-proxy/blob/master/src/oauthHandlers/tokenHandlerStrategyClasses/saveDocumentStrategies/saveDocumentLaunchStrategy.js

This does add a bit of complexity since right now we aren't doing any validation on the provided launch context.

We should check that it is either 

If Base64 encoded json
- If there a key of `patient`, and in that case validate to a string is provided

If Non-base64 encoded json
- Assumed to be the patient, validate that the provided value is a string


If either scenario finds an integer, it should return a 400 error like below.

```
{
    "error": "invalid_launch",
    "error_description": "The provided patient launch must be a string."
}
```

---

- oops I may have advised him to the wrong spot.

---

So instead of the String cast we should throw a 400? 

Also launch should always be an encoding or a JSON like below:

{
  patient: 333
}

---

In the JSON above throw the error?

---

Yeah, something like

```
throw {
      status: 400,
      error: "invalid_request",
      error_description: "The provided patient launch must be a string",
    };
```

or

```
throw {
      status: 400,
      error: "invalid_launch",
      error_description: "The provided patient launch must be a string",
    };
```

---

Thanks Larry!

---

Turns out this did not shift the validation far enough left.

An invalid `launch` technically would first have a CCG access token minted, followed by the saveDocumentStrategy call hitting the validation and returning the error.

We need to check if in the TokenHandlerClient itself prior to the token minting is possible without too much churn around the other flows.